### DISCUSSION TITLE
Berry import time

### DISCUSSION THREAD

**Discussion by trlafleur** [arendst/Tasmota#19633] [General]
In the Berry console, if I `import time`, I get an error message, is time function not supported?
In the Tasmota source tree, I see it used in a number of examples...

~~~
Welcome to the Berry Scripting console. Check the [documentation](https://tasmota.github.io/docs/Berry/).
import time
BRY: Exception> 'import_error' - module 'time' not found
stack traceback:
	<unknown source>: in function `main`

[Tasmota 13.1.0.3 by Theo Arends](https://bit.ly/tasmota)
~~~

**Comment by sfromis** [comment]
Why do you expect `import time` to be available, and what specific time function are you looking for?
https://tasmota.github.io/docs/Berry/#functions-for-time-timers-or-cron

I think that https://github.com/arendst/Tasmota/tree/development/lib/libesp32/berry/examples is generic examples, not for the Tasmota implementation. Other functions found there, like `input('prompt')` are also not for Tasmota.

**Comment by trlafleur** [comment]
Well, it was in the Berry documentation and was in the example in the Berry
source tree in Tasmota source.  Yes there are other options

**Reply by sfromis** [reply]
As an embedded language, operating system dependent features of Berry will generally be tailored for the host platform. Not sure why the `time` module was not included, but it seems that reasonably close matches are:
`time.time()` -> `tasmota.rtc()['local']`
`time.dump(ts)` -> `tasmota.time_dump(ts)`
`time.clock()` -> `tasmota.millis() / 1000.`

**Reply by sfromis** [reply]
It may be possible to enable the `time` module using `BE_USE_TIME_MODULE` in [berry_conf.h](https://github.com/arendst/Tasmota/blob/development/lib/libesp32/berry/default/berry_conf.h), but not sure if it would be workable
